server/app/app.py

```diff
@@ -204,9 +204,11 @@ def __init__(self, app_config: AppConfig):
                 methods=["GET"],
             )
             self.app.add_url_rule(
-                f"/{url_dataroot}/<path:dataset>.cxg/",
+                f"/{url_dataroot}/{CELLGUIDE_CXG_KEY_NAME}/<path:dataset>.cxg/",
```

Comment:
This explicitly ensures that we only capture URLs that for this rule that are prefixed by "cellguide-cxgs/". <path:dataset> was interacting poorly with <string:dataset> because one subsumes the other.